From f89333a5ee935c87a8c3ff7c72e100f59b03afbe Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Thu, 20 Jul 2023 16:32:26 +0200
Subject: [PATCH] Fix parallel build of optee_ta

The symlink must be established prio to building any of the source files
of the lib and WolfSSL, or things will fail:

	TA_CROSS_COMPILE= \
	TA_CPU=cortex-a7 \
	TA_DEV_KIT_DIR=/usr/lib/optee-os/stm32mp15x/export-ta_arm32 \
	CFG_TEE_TA_LOG_LEVEL=2 \
	CFG_FTPM_USE_WOLF=y \
	/usr/bin/make -j 24
make[2]: Entering directory '/<<PKGBUILDDIR>>/Samples/ARM32-FirmwareTPM/optee_ta'
/usr/bin/make -C fTPM CROSS_COMPILE=
make[3]: Entering directory '/<<PKGBUILDDIR>>/Samples/ARM32-FirmwareTPM/optee_ta/fTPM'
Checking symlink to the TPM folder: /<<PKGBUILDDIR>>
Checking symlink to the WolfSSL folder: /<<PKGBUILDDIR>>/external/wolfssl
Establishing symlink.
  CC      ../out/fTPM/platform/Cancel.o
Establishing symlink.
  CC      ../out/fTPM/platform/AdminPPI.o
  CC      ../out/fTPM/platform/Entropy.o
make[3]: *** No rule to make target 'lib/wolf/wolf_symlink/wolfcrypt/src/aes.c', needed by '../out/fTPM/./lib/wolf/wolf_symlink/wolfcrypt/src/aes.o'.  Stop.
make[3]: *** Waiting for unfinished jobs....

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 Samples/ARM32-FirmwareTPM/optee_ta/Makefile    | 9 ++++++++-
 Samples/ARM32-FirmwareTPM/optee_ta/fTPM/sub.mk | 6 ------
 2 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/Samples/ARM32-FirmwareTPM/optee_ta/Makefile b/Samples/ARM32-FirmwareTPM/optee_ta/Makefile
index ddf6784..cef0c6f 100644
--- a/Samples/ARM32-FirmwareTPM/optee_ta/Makefile
+++ b/Samples/ARM32-FirmwareTPM/optee_ta/Makefile
@@ -3,8 +3,15 @@ export V?=0
 
 .PHONY: all
 all:
+	#
+	# Link the required external code into the libraries folder. OP-TEE
+	# build does not work well when accessing anything below the root
+	# directory. Use symlinks to trick it.
+	#
+	$(MAKE) -C fTPM CROSS_COMPILE=$(TA_CROSS_COMPILE) create_lib_symlinks
+	# Run the actual build
 	$(MAKE) -C fTPM CROSS_COMPILE=$(TA_CROSS_COMPILE)
 
 .PHONY: clean
 clean:
-	$(MAKE) -C fTPM clean
\ No newline at end of file
+	$(MAKE) -C fTPM clean
diff --git a/Samples/ARM32-FirmwareTPM/optee_ta/fTPM/sub.mk b/Samples/ARM32-FirmwareTPM/optee_ta/fTPM/sub.mk
index 2d9ba24..4eaf286 100644
--- a/Samples/ARM32-FirmwareTPM/optee_ta/fTPM/sub.mk
+++ b/Samples/ARM32-FirmwareTPM/optee_ta/fTPM/sub.mk
@@ -36,12 +36,6 @@ cflags-y += -Os
 cflags-y += -DNDEBUG
 endif
 
-#
-# Link the required external code into the libraries folder. OP-TEE build
-# does not work well when accessing anything below the root directory. Use
-# symlinks to trick it.
-#
-all: create_lib_symlinks
 clean: clean_lib_symlinks
 
 subdirs-y += lib
-- 
2.35.3

